tests/ftqc/test_parametric_mid_measure.py

```diff
@@ -0,0 +1,426 @@
+        and false condition respectively"""
+
+        with qml.queuing.AnnotatedQueue() as q:
+            qml.RX(1.2, 0)
+            m = qml.measure(0)
+            qml.cond(
+                m == 1,
+                partial(ParametricMidMeasureMP, angle=1.2),
+                partial(ParametricMidMeasureMP, angle=2.4),
+            )(wires=2, plane="XY")
+        original_tape = qml.tape.QuantumScript.from_queue(q)
+        true_diag_gate = ParametricMidMeasureMP(2, angle=1.2, plane="XY").diagonalizing_gates()[0]
+        false_diag_gate = ParametricMidMeasureMP(2, angle=2.4, plane="XY").diagonalizing_gates()[0]
+
+        (new_tape,), _ = diagonalize_mcms(qml.tape.QuantumScript.from_queue(q))
+        assert len(new_tape.operations) == 6
+
+        for idx1, idx2, base_diagonalizing_gate in [
+            (2, 3, true_diag_gate),
+            (4, 5, false_diag_gate),
+        ]:
+            diagonalizing_gate = new_tape.operations[idx1]
+            measurement = new_tape.operations[idx2]
+
+            # conditional diagonalizing gate
+            assert isinstance(diagonalizing_gate, qml.ops.Conditional)
+            assert diagonalizing_gate.base == base_diagonalizing_gate
+            assert diagonalizing_gate.wires == original_tape.operations[2].wires
+
+            # conditional diagonalized ParametricMidMeasureMP
+            assert isinstance(measurement, qml.ops.Conditional)
+            assert measurement.wires == original_tape.operations[2].wires
+            assert isinstance(measurement.base, ParametricMidMeasureMP)
+            assert measurement.base.angle == 0
+            assert measurement.base.plane == "ZY"
+            assert np.allclose(
+                measurement.base.diagonalizing_gates()[0].matrix(), qml.I(0).matrix()
+            )
+
+        # all 4 conditionals rely on the same measurement value
+        for gate in new_tape.operations[3:]:
+            assert gate.meas_val.measurements[0] == new_tape.operations[2].meas_val.measurements[0]
+
+        # diagonalized gates each have processing functions matching their respective cond(mcm)
+        assert (
+            new_tape.operations[2].meas_val.processing_fn
+            == new_tape.operations[3].meas_val.processing_fn
+        )
+        assert (
+            new_tape.operations[4].meas_val.processing_fn
+            == new_tape.operations[5].meas_val.processing_fn
+        )
+        assert (
+            new_tape.operations[2].meas_val.processing_fn
+            != new_tape.operations[4].meas_val.processing_fn
+        )
+
+
+class TestWorkflows:
+    @pytest.mark.parametrize("mcm_method, shots", [("tree-traversal", None), ("one-shot", 10000)])
+    def test_execution(self, mcm_method, shots):
+        """Test that we can execute a QNode with a ParametricMidMeasureMP and produce
+        an accurate result"""
+
+        dev = qml.device("default.qubit", shots=shots)
+
+        @diagonalize_mcms
```

Comment:
Is there a failure without this in place, or just wrong results?
May be worth tracking this as an xfail if so.

Reply:
Wrong results - it will behave as a MCM in the computational basis instead of in the assigned basis